### PR TITLE
mon: fix misc asok commands

### PR DESCRIPTION
**Pull request by liewegas**
In the big asok/tell cleanup, I missed a bunch of these commands that were registered as CLI commands but were really asok/tell commands.

Fixes: https://tracker.ceph.com/issues/42387

### CODE REVIEW

File: src/mon/MonCommands.h [hunk 1]
```diff
@@ -242,20 +242,6 @@ COMMAND("mon ok-to-rm " \
...
 
-COMMAND_WITH_FLAG("mon_status", "report status of monitors", "mon", "r",
-	     FLAG(NOFORWARD))
-COMMAND_WITH_FLAG("sync force " \
-	"name=yes_i_really_mean_it,type=CephBool,req=false " \
-	"name=i_know_what_i_am_doing,type=CephBool,req=false", \
-	"force sync of and clear monitor store", \
-        "mon", "rw", \
-        FLAG(NOFORWARD)|FLAG(DEPRECATED))
-COMMAND_WITH_FLAG("heap " \
```
Comment (tchaikov) [inline src/mon/MonCommands.h]:
```
======================================================================
FAIL: test_ceph_argparse.TestMonitor.test_heap
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/pybind/test_ceph_argparse.py", line 340, in test_heap
    self.assert_valid_command(['heap', 'dump'])
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/pybind/test_ceph_argparse.py", line 53, in assert_valid_command
    assert_not_in(result, [{}, None])
AssertionError: {} unexpectedly found in [{}, None]

```

"heap" command is removed from the registry. so we need to remove the corresponding tests from `test_ceph_argparse.py`.

File: src/mon/MonCommands.h [hunk 2]
```diff
@@ -242,20 +242,6 @@ COMMAND("mon ok-to-rm " \
 	"check whether removing the specified mon would break quorum",
 	"mon", "r")
 
-COMMAND_WITH_FLAG("mon_status", "report status of monitors", "mon", "r",
```
Comment (tchaikov) [inline src/mon/MonCommands.h]:
```
======================================================================
FAIL: test_ceph_argparse.TestMonitor.test_mon_status
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/pybind/test_ceph_argparse.py", line 318, in test_mon_status
    self.assert_valid_command(['mon_status'])
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/pybind/test_ceph_argparse.py", line 53, in assert_valid_command
    assert_not_in(result, [{}, None])
AssertionError: {} unexpectedly found in [{}, None]
```
ditto.

File: src/mon/MonCommands.h [hunk 3]
```diff
@@ -242,20 +242,6 @@ COMMAND("mon ok-to-rm " \
...
-	"name=yes_i_really_mean_it,type=CephBool,req=false " \
-	"name=i_know_what_i_am_doing,type=CephBool,req=false", \
-	"force sync of and clear monitor store", \
-        "mon", "rw", \
-        FLAG(NOFORWARD)|FLAG(DEPRECATED))
-COMMAND_WITH_FLAG("heap " \
-	     "name=heapcmd,type=CephChoices,strings=dump|start_profiler|stop_profiler|release|stats", \
-	     "show heap usage info (available only if compiled with tcmalloc)", \
-	     "mon", "rw", FLAG(NOFORWARD))
-COMMAND("quorum name=quorumcmd,type=CephChoices,strings=enter|exit,n=1", \
```
Comment (tchaikov) [inline src/mon/MonCommands.h]:
```

======================================================================
FAIL: test_ceph_argparse.TestMonitor.test_quorum
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/pybind/test_ceph_argparse.py", line 349, in test_quorum
    self.assert_valid_command(['quorum', 'enter'])
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/pybind/test_ceph_argparse.py", line 53, in assert_valid_command
    assert_not_in(result, [{}, None])
AssertionError: {} unexpectedly found in [{}, None]

======================================================================
FAIL: test_ceph_argparse.TestMonitor.test_sync_force
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/pybind/test_ceph_argparse.py", line 324, in test_sync_force
    '--i-know-what-i-am-doing'])
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/pybind/test_ceph_argparse.py", line 53, in assert_valid_command
    assert_not_in(result, [{}, None])
AssertionError: {} unexpectedly found in [{}, None]
```
ditto.

File: src/mon/MonCommands.h (hunk 1 above)
Comment (sebastian-philipp) [reply src/mon/MonCommands.h]: There is a way to find these at make check, downside would be the need to add the required frameworks and generated APIs.

**Comment by liewegas** [comment]
retest this please

**Pull request merged**